src/language-server/config/config.ts

```diff
@@ -197,24 +197,20 @@ export function parseApolloConfig(
         });
       }
     }
-    Debug.error(
-      fromZodError(parsed.error, {
-        prefix: "Error parsing config file:",
-        prefixSeparator: "\n",
-        issueSeparator: ";\n",
-        unionSeparator: "\n  or\n",
-      }).toString(),
-    );
-    return null;
+    throw fromZodError(parsed.error, {
+      prefix: `Error parsing config file ${configURI?.fsPath}:`,
+      prefixSeparator: "\n",
+      issueSeparator: ";\n",
+      unionSeparator: "\n  or\n",
+    });
```

Comment:
There was a full mechanism for handling thrown errors, but instead a lot of places logged errors instead. Defaulting to throwing in most places now.